### PR TITLE
Update scenario executive summary

### PR DESCRIPTION
**Pull request by MonikaFu**
Updating scenario source and scenario for executive summary calculations. Aligned with what the team decided to use.

- [x] depends on https://github.com/RMI-PACTA/pacta.executive.summary/pull/325
- [x] depends on https://github.com/RMI-PACTA/pacta.executive.summary/pull/327

### CODE REVIEW

**Comment by MonikaFu** [comment]
This PR is dependent on: https://github.com/RMI-PACTA/pacta.executive.summary/pull/325 that is why I marked it as 'draft'.

**Comment by cjyetman** [comment]
@MonikaFu https://github.com/RMI-PACTA/pacta.executive.summary/pull/325 has been merged, but this is still in draft mode. Is it ready for review?

**Comment by MonikaFu** [comment]
It's probably best after we also merge this one: https://github.com/RMI-PACTA/pacta.executive.summary/pull/327#discussion_r1620413578. Then we have the right datasets for calculating the scores.

**Comment by MonikaFu** [comment]
It's ready for review now.

**Review by jdhoffa — DISMISSED** [review]
lgtm!
But let's wait then for the CI/CD to run and see what it looks like :-)

**Comment by jdhoffa** [comment]
I'm seeing loads of "an error occured when generating this plot" in the CI/CD results, e.g.:
https://pactadatadev.blob.core.windows.net/ghactions-workflow-transition-monitor-results-reports/rmi_pacta_2023q4_pa2024ch-20240530T150052Z/EN/bank/1/working_dir/50_Outputs/rmi_pacta_2023q4_pa2024ch/executive_summary/template.pdf

**Review by cjyetman — APPROVED** [review]
this seems ok to me, but I have no idea how this affects the plots in the exec summary

**Comment by jdhoffa** [comment]
I manually re-triggered Docker build and test off this branch here:
https://github.com/RMI-PACTA/workflow.transition.monitor/actions/runs/9314690620

Wondering if maybe it didn't pick up the latest changes in `pacta.executive.summary`. I will link the built test ES pdfs here once they are built, but I would suggest that we don't merge this PR until we get the plots we expect.

**Comment by MonikaFu** [comment]
Agreed @jdhoffa. It seems that the docker build is not always triggered when changes are made to other PRs. It happened before.

**Comment by cjyetman** [comment]
> I manually re-triggered Docker build and test off this branch here: https://github.com/RMI-PACTA/workflow.transition.monitor/actions/runs/9314690620
> 
> Wondering if maybe it didn't pick up the latest changes in `pacta.executive.summary`. I will link the built test ES pdfs here once they are built, but I would suggest that we don't merge this PR until we get the plots we expect.

FYI... usually if a PR is created (technically, if an action runs) *before* a change in another repo is available on its main branch, re-running the action will *not* pickup new changes on the other repo... it seems to cache whatever was there the first time the action was run. So, if that situation is relevant here, you may want to open a new PR with the same changes here and see if the action succeeds with that.

**Comment by AlexAxthelm** [comment]
> > I manually re-triggered Docker build and test off this branch here: https://github.com/RMI-PACTA/workflow.transition.monitor/actions/runs/9314690620
> > Wondering if maybe it didn't pick up the latest changes in `pacta.executive.summary`. I will link the built test ES pdfs here once they are built, but I would suggest that we don't merge this PR until we get the plots we expect.
> 
> FYI... usually if a PR is created (technically, if an action runs) _before_ a change in another repo is available on its main branch, re-running the action will _not_ pickup new changes on the other repo... it seems to cache whatever was there the first time the action was run. So, if that situation is relevant here, you may want to open a new PR with the same changes here and see if the action succeeds with that.

More specificly, the behavior I've observed (but not seen documented anywhere) is For a given `run` of a workflow, all GH-related state information (such as the `HEAD` for workflows in the RMI-PACTA/actions repo, or the commit that is checked out with the `actions/checkout` step) is preserved across `attempts` ("re-run failed jobs"), but anything external (Azure resources, for example) are still "live".

You can trigger a new "run" (with new caching of state) by pushing a new commit. For example, see this PR (https://github.com/RMI-PACTA/workflow.pacta/pull/59), where I add and delete trivial whitespace in order to explicitly trigger new `run`s.

**Comment by jdhoffa** [comment]
The issues seems to persist any case:
See this report:
 https://pactadatadev.blob.core.windows.net/ghactions-workflow-transition-monitor-results-reports/rmi_pacta_2023q4_pa2024ch-20240531T081134Z/EN/bank/1/working_dir/50_Outputs/rmi_pacta_2023q4_pa2024ch/executive_summary/template.pdf
from this branch:
https://github.com/RMI-PACTA/workflow.transition.monitor/pull/320#issuecomment-2141540005

Screenshot 2024-05-31 at 11 24 03

**Comment by MonikaFu** [comment]
This is very strange because when I run the PR locally with all the latest versions of packages on `main` I get the plots showing up.

**Comment by AlexAxthelm** [comment]
> This is very strange because when I run the PR locally with all the latest versions of packages on `main` I get the plots showing up.

Are you using the same testing data?

**Comment by MonikaFu** [comment]
> > This is very strange because when I run the PR locally with all the latest versions of packages on `main` I get the plots showing up.
> 
> Are you using the same testing data?

Now yes I believe and I still get some plots to appear although some do not. But I don't get the same pdf as @jdhoffa with no plots generated.

**Comment by MonikaFu** [comment]
First thing I discovered is that in the indices data the scenario name we want to be using from `WEO2023` is `NZE_2050` and not `NZE`.

**Comment by AlexAxthelm** [comment]
> Until now this did not lead to issues but given that now indices are generated dynamically it might be beneficial to refactor the code of ES to use the same indices as the rest of `workflow.transition. monitor`.

Do we know why we used different files in the first place? I'm generally in favor of integrating them though.

**Comment by jdhoffa** [comment]
> > Until now this did not lead to issues but given that now indices are generated dynamically it might be beneficial to refactor the code of ES to use the same indices as the rest of `workflow.transition. monitor`.
> 
> Do we know why we used different files in the first place? I'm generally in favor of integrating them though.

For the record, I did know/ notice that indices files had to be copied into the `user_results` folder, but I have no idea why they were (e.g. if it was done intentionally, or just a tech debt artifact). @cjyetman do you have any insights there?

**Comment by MonikaFu** [comment]
I'm pretty sure @cjyetman does not have insights here. It is an artifact of ES code. I guess it is easier to access them this way from within the ES template but I'm not sure why the person who wrote this code did not access indices the same way they access pacta results.

**Comment by jdhoffa** [comment]
Are the indices files in the `user_data` folder even used??
The indices data that is passed to `prep_executive_summary` in `web_tool_3` are the identical data that gets passed to `create_interactive_report`:
https://github.com/RMI-PACTA/workflow.transition.monitor/blob/26473e52f666201fd3fbab253bc3492c5e4884cf/web_tool_script_3.R#L422-L423

https://github.com/RMI-PACTA/workflow.transition.monitor/blob/26473e52f666201fd3fbab253bc3492c5e4884cf/web_tool_script_3.R#L360-L361

which are the ones prepared directly by `workflow.prepare.pacta.indices`:
https://github.com/RMI-PACTA/workflow.transition.monitor/blob/26473e52f666201fd3fbab253bc3492c5e4884cf/web_tool_script_3.R#L170-L172

@MonikaFu can you try removing the indices data from `user_data` and running `web_tool_3` and see if anything fails?

**Comment by MonikaFu** [comment]
@jdhoffa I'm pretty sure I did in the past and it failed.

**Comment by jdhoffa** [comment]
Ah yeah, here's the problem:
https://github.com/RMI-PACTA/pacta.executive.summary/blob/a41ab4d677d3a9fb911a4b4eda0fa5dbbe3aff6f/R/prep_data_executive_summary.R#L247-L269

**Comment by jdhoffa** [comment]
That makes sense then. 

The reason that it is copied in is because we need the `Indices_*_audit` and `Indices_*_emissions` datasets, but only the `Indices_*_results` data ever actually gets loaded into `web_tool_script_3` (as it stands)

**Comment by MonikaFu** [comment]
> Ah yeah, here's the problem: https://github.com/RMI-PACTA/pacta.executive.summary/blob/a41ab4d677d3a9fb911a4b4eda0fa5dbbe3aff6f/R/prep_data_executive_summary.R#L247-L269

Right! It's the indices emissions files which are only used in ES I believe.

**Comment by jdhoffa** [comment]
Ok, I manually updated the `user_data` test data directories on Azure to contain the latest Indices data (brute force way I know, but I just want to make sure THIS PR is fine).

I will re-trigger CI/CD and see if it builds correctly now. 

We can deal with the problem of index copying later.

**Comment by jdhoffa** [comment]
Lol now we are getting a totally different Azure related failure 🤯 
Screenshot 2024-05-31 at 16 29 25

I think that's all I have in me for a Friday afternoon, but @AlexAxthelm let's look into this next week (or more realistically the week after)

**Comment by AlexAxthelm** [comment]
Opened https://github.com/Azure/login/issues/453 to help understand what's going on with those. It looks like the `azure/login` action changed its behavior, but I'm not sure why.

**Comment by cjyetman** [comment]
> For the record, I did know/ notice that indices files had to be copied into the `user_results` folder, but I have no idea why they were (e.g. if it was done intentionally, or just a tech debt artifact). @cjyetman do you have any insights there?

no idea, but sounds like you figured it out above

**Comment by jdhoffa** [comment]
Ok @MonikaFu seems like the Docker build is running again, but the ES still has persistent issues: 
https://pactadatadev.blob.core.windows.net/ghactions-workflow-transition-monitor-results-reports/rmi_pacta_2023q4_pa2024ch-20240603T130203Z/EN/bank/1/working_dir/50_Outputs/rmi_pacta_2023q4_pa2024ch/executive_summary/template.pdf

**Comment by MonikaFu** [comment]
As I said, when I run locally with the latest versions of packages and the same data as on docker I get most of the plots (apart from the indices issue). So I'm not sure if it's an issue of this PR or data/docker issue.

**Comment by MonikaFu** [comment]
[template.pdf](https://github.com/user-attachments/files/15535076/template.pdf)

Here what I get locally.

**Comment by jdhoffa** [comment]
Just ran it locally with the latest Azure data, and I also get the same "error no plot" issue. 
Screenshot 2024-06-03 at 16 27 34

So evidently there is some difference between our local set-ups.

File: web_tool_script_3.R
Comment (jdhoffa) [inline web_tool_script_3.R:399]:
```suggestion
      scenario_selected = "NZE_2050",
```

The April 24th run of data prep has this specification for scenario name:
Screenshot 2024-06-03 at 16 42 45

Also in line with what the config file expects:
https://github.com/RMI-PACTA/workflow.transition.monitor/blob/7591a74a826a02fa12b86bb41ef8b8ea71931355/parameter_files/ProjectParameters_PA2024CH.yml#L14-L15

File: web_tool_script_3.R
Comment (jdhoffa) [inline web_tool_script_3.R:442]:
```suggestion
    scenario_selected = "NZE_2050",
```

File: web_tool_script_3.R
Comment (jdhoffa) [reply web_tool_script_3.R:399]: This is also how it is specified in `pacta.portfolio.report`:
https://github.com/RMI-PACTA/pacta.portfolio.report/blob/0c1837ea39e4f9acfffdacb53bd049c11048bb3c/inst/js/trajectory_alignment.js#L423-L428

**Comment by jdhoffa** [comment]
FYI @MonikaFu I think it still won't work until https://github.com/RMI-PACTA/pacta.executive.summary/pull/330 is merged

**Comment by MonikaFu** [comment]
It depends on the `scenario_thresholds` dataset I believe so yes, correct, I don't expect it to work without that PR.

**Comment by jdhoffa** [comment]
Re-triggered CI/CD, we shall see what happens there. 

Indices data has already been updated on Azure `user_results`, so that should no longer be a problem

File: web_tool_script_3.R
Comment (cjyetman) [reply web_tool_script_3.R]: for future reference, the scenario-source pairs are included in the manifest
``` r
jsonlite::read_json("~/Downloads/2023Q4_20240527T092929Z/manifest.json")$parameters$output_stats$source_scenario_pairs
```

Comment (cjyetman) [reply web_tool_script_3.R]: or better...
``` r
manifest <- jsonlite::fromJSON("~/Downloads/2023Q4_20240527T092929Z/manifest.json")
manifest$parameters$output_stats$source_scenario_pairs
#>   scenario_source  scenario
#> 1        GECO2023      1.5C
#> 2        GECO2023   NDC-LTS
#> 3        GECO2023 Reference
#> 4         ISF2023     1.5°C
#> 5         WEO2023       APS
#> 6         WEO2023  NZE_2050
#> 7         WEO2023     STEPS
```

**Comment by cjyetman** [comment]
> Re-triggered CI/CD, we shall see what happens there.
> 
> Indices data has already been updated on Azure `user_results`, so that should no longer be a problem

FYI, I think @AlexAxthelm clarified that re-triggering (telling GitHub to re-run) will use cached versions of external repos (therefor, not pick up recent changes to external repos), but adding a new commit to this PR will clear the cache and re-run with most recent version of external repos.

**Comment by MonikaFu** [comment]
@cjyetman just added and removed a line of code, let's see if this works.

**Comment by MonikaFu** [comment]
@cjyetman yes, indeed, it looks like it. Looking into it now and trying to add missing pieces with https://github.com/RMI-PACTA/pacta.executive.summary/pull/331.

**Comment by cjyetman** [comment]
@MonikaFu do you have any idea why things run as expected on your machine but not here? To me, that's a major source of confusion making this very difficult to resolve. I think we should try to get things aligned before committing to various fixes that don't actually work.

**Comment by jdhoffa** [comment]
> @MonikaFu do you have any idea why things run as expected on your machine but not here? To me, that's a major source of confusion making this very difficult to resolve. I think we should try to get things aligned before committing to various fixes that don't actually work.

My hunch would be that @MonikaFu's local data was out of sync, since locally she was able to produce a report prior to the `NZE -> NZE_2050` commits

**Comment by jdhoffa** [comment]
@MonikaFu what is the status of this?

**Review by cjyetman — APPROVED** [review]

**Pull request merged**
